### PR TITLE
Fixes #28857 - protect the Intl from code duplications

### PR DESCRIPTION
**Pull request by sharvit**
Protect the `Intl` from code duplications.
Also, provide a method to force singletons and migrate other places so it would be easier to replace it when/if needed.

### CODE REVIEW

File: webpack/assets/javascripts/react_app/redux/index.js [hunk 1]
```diff
@@ -25,8 +26,6 @@ const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
 export const generateStore = () =>
   createStore(reducers, composeEnhancers(applyMiddleware(...middleware)));
 
-window.tfm_redux_store = window.tfm_redux_store || generateStore();
-
-const store = window.tfm_redux_store;
+const store = forceSingleton('redux_store', () => generateStore());
```
Comment (tbrisker) [inline webpack/assets/javascripts/react_app/redux/index.js]: Would this work the same?
```suggestion
const store = forceSingleton('redux_store', generateStore);
```

File: webpack/assets/javascripts/react_app/common/forceSingleton.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+ * WARNING: Code duplications happen because of an issue with the build process,
+ *          so this method might be removed once the issue would be fixed.
+ *          See: https://projects.theforeman.org/issues/27195
+ *
+ * @param  {String}   key    A unique-key to save the instance.
+ * @param  {Function} create A function to create an instance.
+ * @return {*}               Single Instance,
+ *                           returned by the create method or from the cache.
+ */
+const forceSingleton = (key, create) => {
```
Comment (tbrisker) [inline webpack/assets/javascripts/react_app/common/forceSingleton.js:13]: I guess in many cases the default would be an empty object, would it make sense to use:
```suggestion
const forceSingleton = (key, create = () => ({})) => {
```
and only pass the second attribute if needed?

**Review by tbrisker — COMMENTED** [review]
a couple of inline comments, also i think you missed the component registry

File: webpack/assets/javascripts/react_app/common/forceSingleton.js (hunk 2 above)
Comment (sharvit) [reply webpack/assets/javascripts/react_app/common/forceSingleton.js:13]: IMO it's a hack and I'm not sure about its place once we will fix the build system so
maybe we shouldn't make it too friendly?

File: webpack/assets/javascripts/react_app/redux/index.js (hunk 1 above)
Comment (sharvit) [reply webpack/assets/javascripts/react_app/redux/index.js]: Yes, replaced

**Review by tbrisker — APPROVED** [review]
Thanks @sharvit !

**Pull request merged**